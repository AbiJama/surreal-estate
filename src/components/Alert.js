import React from 'react';
import "../styles/Alert.css";

const Alert = ({ message, success }) => {
  if (!message) return null;

  return (
    <div className={`alert alert-${success ? "success" : "error"}`}>
      {message}
    </div>
  );
};

export default Alert;
