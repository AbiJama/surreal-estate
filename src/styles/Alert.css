.alert {
    padding: 10px;
    margin: 10px 0;
    border: 1px solid #ccc;
    border-radius: 5px;
  }
  
  .success {
    background-color: #a5d6a7; /* Green background for success */
  }
  
  .error {
    background-color: #ef9a9a; /* Red background for error */
  }