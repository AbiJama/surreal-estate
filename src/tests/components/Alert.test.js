import React from "react";
import { render } from "@testing-library/react";
import Alert from "../../components/Alert";

test("displays an error message", () => {
  const { getByText } = render(<Alert message="Error!" />);

  expect(getByText(/Error/)).toBeInTheDocument();

  expect(getByText(/Error/).textContent).toBe("Error!");
});

test("displays a success message", () => {
  const { getByText } = render(<Alert message="Success!!!!" success />);

  expect(getByText(/Success/).textContent).toBe("Success!!!!");
});

test("renders correctly with an error message", () => {
  const { asFragment } = render(<Alert message="Error!" />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});

test("renders correctly with a success message", () => {
  const { asFragment } = render(<Alert message="Success!!!!" success />);
  const alert = asFragment();

  expect(alert).toMatchSnapshot();
});

test("does not render an error or a success message if message prop is empty", () => {
    const { asFragment } = render(<Alert message="" />);
    const alert = asFragment();
  
    expect(alert).toMatchSnapshot();
  });
