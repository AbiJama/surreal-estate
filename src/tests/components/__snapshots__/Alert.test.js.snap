// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render an error or a success message if message prop is empty 1`] = `<DocumentFragment />`;

exports[`renders correctly with a success message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!!!
  </div>
</DocumentFragment>
`;

exports[`renders correctly with an error message 1`] = `
<DocumentFragment>
  <div
    class="alert alert-error"
  >
    Error!
  </div>
</DocumentFragment>
`;
